x-pack/plugins/uptime/public/components/overview/monitor_list/filter_status_button.tsx

```diff
@@ -9,29 +9,38 @@ import React from 'react';
 import { useUrlParams } from '../../../hooks';
 
 export interface FilterStatusButtonProps {
-  content: string;
+  content: string | JSX.Element;
   dataTestSubj: string;
   isDisabled?: boolean;
+  isActive: boolean;
   value: string;
   withNext: boolean;
+  color?: string;
 }
 
 export const FilterStatusButton = ({
   content,
   dataTestSubj,
   isDisabled,
+  isActive,
   value,
+  color,
   withNext,
 }: FilterStatusButtonProps) => {
   const [getUrlParams, setUrlParams] = useUrlParams();
   const { statusFilter: urlValue } = getUrlParams();
+
   return (
     <EuiFilterButton
+      color={(isActive ? color : undefined) as any}
       data-test-subj={dataTestSubj}
-      hasActiveFilters={urlValue === value}
+      hasActiveFilters={isActive}
       isDisabled={isDisabled}
       onClick={() => {
-        const nextFilter = { statusFilter: urlValue === value ? '' : value, pagination: '' };
+        const nextFilter = {
+          statusFilter: urlValue === value || value === 'all' ? '' : value,
```

Comment:
Per your comment elsewhere, I think you're right that using a hook here is OK, but I think there's something off about this logic. AFAICT we don't ever set the value to `'all'` but rather to `''`. 

Also, do we have to use similar logic elsewhere in the app? It feels like a smell that this specific component is examining strings at this level of detail to determine this mapping of URL values.